newsroom/history.py

```diff
@@ -111,7 +112,48 @@ def create_media_history_record(self, item, association_name, action, user, sect
         except (werkzeug.exceptions.Conflict, pymongo.errors.BulkWriteError):
             pass
 
-    def log_media_download(self, item_id, media_id):
+    def _find_association(self, item, media_id):
+        """
+        Find the matching media association in the item
+        :param item: item object
+        :param media_id: ID of the media
+        :return: tuple (name, association) or a 404
+        """
+        for name, association in (item.get('associations') or {}).items():
+            for rendition in association.get("renditions"):
```

Comment:
It's plausible that `renditions` here could not be defined, in which you would get the following exception raised:

```bash
TypeError: 'NoneType' object is not iterable
```